addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -368,65 +370,86 @@ public Transfer generateGroupFileContent( final Group group, final List<Artifact
             logger.trace( "Metadata file exists for group {} of path {}, no need to regenerate.", group.getKey(), path );
             return target;
         }
-        try
-        {
-            target.lockWrite();
 
-            final Metadata md = generateGroupMetadata( group, members, path );
-            if ( md != null )
+        if ( mergerLock.tryLock() )
+        {
+            try
```

Comment:
We need to store the lock state output from tryLock() and only try to unlock if it worked. We might need to check that boolean before proceeding with this operation too. See what I've done in PromotionManager (it might only be partially complete, WRT proceeding through the try/catch without checking the boolean):

https://github.com/Commonjava/indy/blob/c78a7856dd3b4683f7d2d32937bb3fae4b59d0d9/addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PromotionManager.java#L511-L511